akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -165,6 +295,14 @@ abstract class GraphStageLogic private[stream] (inCount: Int, outCount: Int) {
   }
 
   /**
+   * Requests an element on the given port unless the port is already closed.
+   * Calling this method twice before an element arrived will fail.
+   * There can only be one outstanding request at any given time. The method [[hasBeenPulled()]] can be used
+   * query whether pull is allowed to be called or not.
+   */
+  final protected def tryPull[T](in: Inlet[T]): Unit = if (!isClosed(in)) pull(in)
```

Comment:
I lifted it from multiple improvised use sites, which tells me that there is utility in having it handy.